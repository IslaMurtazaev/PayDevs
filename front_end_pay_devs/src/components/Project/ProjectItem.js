import React, { Component } from "react";
import { connect } from "react-redux";
import { projectActions } from "../../actions/project";
import { Link } from "react-router-dom";
import Tasks from "../../components/Task/Tasks";
import HourPayments from "../../components/HourPayment/HourlyRates";
import MonthRates from "../MonthPayment/MonthlyRates";

class ProjectItem extends Component {
  componentDidMount() {
    if (!this.props.project.title) this.props.getProject();
  }

  render() {
    let { project, getTotal, removeProject } = this.props;

    let sessionsType;
    let type_of_payment;
    switch (project.type_of_payment) {
      case "M_P":
        type_of_payment = "Monthly";
        sessionsType = <MonthRates projectId={project.id} />;
        break;
      case "H_P":
        type_of_payment = "Hourly";
        sessionsType = <HourPayments projectId={project.id} />;
        break;
      case "T_P":
        type_of_payment = "Taskly";
        sessionsType = <Tasks projectId={project.id} />;
        break;
      default:
        break;
    }

    return (
      <div className="container">
        <h1 className="projectTitle">{project.title}</h1>
        <div className="properties">
          {project.description ? (
            <h4 className="projectDescription">
              <span className="property">Description: </span>
              {project.description}
            </h4>
          ) : null}
          {project.start_date ? (
            <h4 className="projectStartDate">
              <span className="property">Start date: </span>
              {new Date(project.start_date).toDateString()}
            </h4>
          ) : null}
          {project.end_date ? (
            <h4 className="projectEndDate">
              <span className="property">End date: </span>
              {new Date(project.end_date).toDateString()}
            </h4>
          ) : null}
          <h4 className="projectTypeOfPayment">
            <span className="property">Type of payment:</span> {type_of_payment}
          </h4>
          <h4 className="projectStatus">
            <span className="property">Status:</span>
            {project.status ? "" : "not"} active
          </h4>
        </div>

        <div className="btn-group-horizontal">
          <button
            type="button"
            className="btn btn-success btn-lg"
            onClick={() => getTotal(project.id)}
          >
            Total
          </button>

          <Link to={`${project.id}/update`}>
            <button type="button" className="btn btn-warning btn-lg">
              Update project
            </button>
          </Link>

          <button
            type="button"
            className="removeProject btn btn-danger btn-lg"
            onClick={() => removeProject(project.id)}
          >
            Delete project
          </button>
        </div>

        <div className="session">{sessionsType}</div>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  project: state.project
    ? state.project
    : state.projects.find(project => project.id === +ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  removeProject: id => {
    dispatch(projectActions.remove(id));
  },
  getTotal: id => {
    dispatch(projectActions.getTotal(id));
  },
  getProject: () => {
    dispatch(projectActions.get(+ownProps.match.params.id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectItem);
