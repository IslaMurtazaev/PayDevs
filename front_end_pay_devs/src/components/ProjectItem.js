import React, { Component } from "react";
import { connect } from "react-redux";
import { projectActions } from "../actions/project";
import { Redirect } from "react-router";

class ProjectItem extends Component {
  onClick(id) {
    this.props.onGetAllProjects(id);
  }

  onClickTotal(id){
    this.props.onGetTotal(id);
  }

  render() {
    let project = this.props.project;
    if (!Object.keys(project).length) return <Redirect from="/project/:id" to="/" />;

    let type_of_payment;
    switch(project.type_of_payment) {
      case "M_P":
        type_of_payment = "Monthly";
        break;
      case "H_P":
        type_of_payment = "Hourly";
        break;
      case "T_P":
        type_of_payment = "Taskly";
        break;
      default:
        break;
    }
    return (
        <div className="container">
        <h3>Title: {project.title}</h3>
        {project.description ? (
          <h4>Description: {project.description}</h4>
        ) : null}
        {project.start_date ? (
          <h4>Start-date: {new Date(project.start_date).toDateString()}</h4>
        ) : null}
        {project.end_date ? (
          <h4>End-date: {new Date(project.end_date).toDateString()}</h4>
        ) : null}
        <h4>Type of payment: {type_of_payment}</h4>
        <h4>Status: {project.status ? "" : "not"} active</h4>
        <button className="btn btn-danger" onClick={this.onClickTotal.bind(this, project.id)}>
          Total
        </button>
        <button className="btn btn-danger" onClick={this.onClick.bind(this, project.id)}>
          DELETE PROJECT
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  let project = state.projects.find(
    product => product.id === Number(ownProps.match.params.id)
  );
  return {
    project: project ? project : state.project
  };
};

export default connect(
  mapStateToProps,
  dispatch => ({
    onGetAllProjects: id => {
      dispatch(projectActions.deleteProject(id));
    },
    onGetTotal:(id)=>{
      dispatch(projectActions.getTotal(id));
    }
  })
)(ProjectItem);
