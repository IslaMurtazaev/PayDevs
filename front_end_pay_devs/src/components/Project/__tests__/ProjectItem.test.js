import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ProjectItem from "../ProjectItem";

describe("<ProjectItem />", () => {
  const initialState = {
    project: {
      title: "PayDevs",
      description: "blablabla",
      start_date: "2018-08-24T00:00:00.000Z",
      end_date: "2018-09-30T10:00:00.000Z",
      type_of_payment: "M_P",
      status: true
    }
  };
  const mockStore = configureStore()

  it("renders without errors", () => {
    // let store = mockStore(initialState);
    // let container = shallow(<ProjectItem store={store} />);
    expect(1).toBe(1);
  });
});
