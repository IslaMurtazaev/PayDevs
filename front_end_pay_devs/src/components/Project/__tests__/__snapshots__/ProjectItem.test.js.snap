// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectItem /> matches previous snap 1`] = `
<div
  className="container"
>
  <h1
    className="projectTitle"
  >
    PayDevs
  </h1>
  <div
    className="properties"
  >
    <h4
      className="projectDescription"
    >
      <span
        className="property"
      >
        Description: 
      </span>
      blablabla
    </h4>
    <h4
      className="projectStartDate"
    >
      <span
        className="property"
      >
        Start date: 
      </span>
      Fri Aug 24 2018
    </h4>
    <h4
      className="projectEndDate"
    >
      <span
        className="property"
      >
        End date: 
      </span>
      Sun Sep 30 2018
    </h4>
    <h4
      className="projectTypeOfPayment"
    >
      <span
        className="property"
      >
        Type of payment:
      </span>
       
      Monthly
    </h4>
    <h4
      className="projectStatus"
    >
      <span
        className="property"
      >
        Status:
      </span>
       active
    </h4>
  </div>
  <div
    className="btn-group-horizontal"
  >
    <button
      className="btn btn-success btn-lg"
      onClick={[Function]}
      type="button"
    >
      Total
    </button>
    <Link
      replace={false}
      to="1/update"
    >
      <button
        className="btn btn-warning btn-lg"
        type="button"
      >
        Update project
      </button>
    </Link>
    <button
      className="removeProject btn btn-danger btn-lg"
      onClick={[Function]}
      type="button"
    >
      Delete project
    </button>
  </div>
  <div
    className="session"
  >
    <Connect(MonthlyRates)
      projectId={1}
    />
  </div>
</div>
`;
